filament/src/components/RenderableManager.cpp

```diff
@@ -253,6 +253,11 @@ RenderableManager::Builder::Result RenderableManager::Builder::build(Engine& eng
             continue;
         }
 
+        // we want a feature level violation to be a hard error (exception if enabled, or crash)
+        ASSERT_PRECONDITION(upcast(engine).hasFeatureLevel(material->getFeatureLevel()),
+                "Material \"%s\" has feature level %u which is not supported this Engine",
```

Comment:
Maybe rephrase and add the current feature level of the engine:
```
Material xyz has feature level X which is not supported. This Engine instance supports feature level Y.
```
Same for `setMaterialInstanceAt`.